api/v1alpha1/stage_types.go

```diff
@@ -81,157 +67,46 @@ const (
 type Stage struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-	// Spec describes the sources of material used by the Stage and how
-	// to incorporate newly observed materials into the Stage.
+	// Spec describes sources of Freight used by the Stage and how to incorporate
+	// Freight into the Stage.
 	//
 	//+kubebuilder:validation:Required
 	Spec *StageSpec `json:"spec"`
-	// Status describes the most recently observed Freight as well as the Stage's
-	// current and recent Freight.
+	// Status describes the Stage's current and recent Freight, health, and more.
 	Status StageStatus `json:"status,omitempty"`
 }
 
 func (s *Stage) GetStatus() *StageStatus {
 	return &s.Status
 }
 
-// StageSpec describes the sources of material used by a Stage and how to
-// incorporate newly observed materials into the Stage.
+// StageSpec describes the sources of Freight used by a Stage and how to
+// incorporate Freight into the Stage.
 type StageSpec struct {
-	// Subscriptions describes the Stage's sources of material. This is a
-	// required field.
+	// Subscriptions describes the Stage's sources of Freight. This is a required
+	// field.
 	//
 	//+kubebuilder:validation:Required
 	Subscriptions *Subscriptions `json:"subscriptions"`
-	// PromotionMechanisms describes how to incorporate newly observed materials
-	// into the Stage. This is an optional field as it is sometimes useful to
-	// aggregates available Freight from multiple upstream Stages without
-	// performing any actions. The utility of this is to allow multiple downstream
-	// Stages to be able to subscribe to a single upstream Stage where they may
-	// otherwise have subscribed to multiple upstream Stages.
+	// PromotionMechanisms describes how to incorporate Freight into the Stage.
+	// This is an optional field as it is sometimes useful to aggregates available
+	// Freight from multiple upstream Stages without performing any actions. The
+	// utility of this is to allow multiple downstream Stages to subscribe to a
+	// single upstream Stage where they may otherwise have subscribed to multiple
+	// upstream Stages.
 	PromotionMechanisms *PromotionMechanisms `json:"promotionMechanisms,omitempty"`
 }
 
-// Subscriptions describes a Stage's sources of material.
+// Subscriptions describes a Stage's sources of Freight.
 type Subscriptions struct {
-	// Repos describes various sorts of repositories a Stage uses as sources of
-	// material. This field is mutually exclusive with the UpstreamStages field.
-	Repos *RepoSubscriptions `json:"repos,omitempty"`
-	// UpstreamStages identifies other Stages as potential sources of material
+	// Warehouse is a subscription to a Warehouse. This field is mutually
+	// exclusive with the UpstreamStages field.
+	Warehouse string `json:"warehouse,omitempty"`
```

Comment:
To future proof this, we may need to make this more like:

```
  warehouseRef:
    name: xxxx
```

Reply:
Just making a note here that when we spoke on Zoom, we agreed to think about this more and tackle it in a follow-up if we deem it necessary.

Reply:
I will open a separate issue for this so we don't lose track of it.

Reply:
#955